### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,138 @@
...
+
+export const newToDos: ToDo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setToDo] = useState(newToDos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('0');
```
Comment (silchyn) [inline src/App.tsx:27]: So initially selected user is `'0'`?

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,138 @@
...
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
```
Comment (silchyn) [inline src/App.tsx:29]:
```suggestion
  const [isTitleValid, setIsTitleValid] = useState(false);
  const [isUserValid, setIsUserValid] = useState(false);
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,138 @@
...
+
+    if (!user) {
+      setUserError(true);
+    }
+
+    if (!user || !isTitleValid) {
+      return;
+    }
+
+    const nextToDoId = (toDoList: ToDo[]) => (
```
Comment (silchyn) [inline src/App.tsx:59]:
```suggestion
    const getNextToDoId = (toDoList: ToDo[]) => (
```

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,138 @@
...
+  )) || null;
+}
+
+export const newToDos: ToDo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setToDo] = useState(newToDos);
```
Comment (dinaminahi) [inline src/App.tsx:25]:
```suggestion
  const [todos, setTodos] = useState(newToDos);
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,138 @@
...
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<string>('Choose a user');
```
Comment (dinaminahi) [inline src/App.tsx:27]: Here you are setting type for the selectedUser variable, but you are not setting it for all the others. Please, be consistent, either set it too all or to nothing

**Review by dinaminahi — APPROVED** [review]
Well done 👍

**Pull request closed without merging**